challenge_8/python/mjuiuc/solution.py

```diff
@@ -0,0 +1,23 @@
+from listNode import listNode
+
+
+def solution(head):
+	lop = dict()
+	newHead = listNode()
+	temp = head
+	temp2 = newHead
+	# copy the list first
+	while temp != None:
+		temp2.val = temp.val
+		lop[temp2.val] = temp2
+		temp2.next = listNode()
```

Comment:
Yea the end of the list is just an empty pointer. Sentinel nodes do exist though, and technically if there is no value at next it will return None anyways. Python is too smart to let me be stupid lol. You're right though it's not defined and it should be.